components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java

```diff
@@ -1237,7 +1237,10 @@ public String exportSPApplicationFromAppID(String applicationId, boolean exportS
                     "for the provided id: " + applicationId);
         }
         String appName = application.getApplicationName();
-        return exportSPApplication(appName, exportSecrets, tenantDomain);
+        startTenantFlow(tenantDomain);
```

Comment:
shall we start the tenant flow in the try block
and end the tenant flow in a finally block?